### ISSUE TITLE
MS IME on windows 10 will make gpui panic

### ISSUE THREAD

**Issue by shenjackyuanjie** [zed-industries/zed#12563]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

when using MS IME on windows 10, it will make the gpui's hello world panic instantly
```
zed\crates\gpui [ HEAD] [📦 v0.1.0][🦀 v1.78.0]
❯ cargo run --example hello_world
warning: V:\githubs\zed\Cargo.toml: unused manifest key: workspace.lints.rust.unexpected_cfgs.check-cfg
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 1.45s
     Running `V:\githubs\zed\target\debug\examples\hello_world.exe`
thread 'main' panicked at crates\gpui\src\platform\windows\platform.rs:142:14:
called `Option::unwrap()` on a `None` value
stack backtrace:
   0: std::panicking::begin_panic_handler
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library\std\src\panicking.rs:645
   1: core::panicking::panic_fmt
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library\core\src\panicking.rs:72
   2: core::panicking::panic
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library\core\src\panicking.rs:145
   3: core::option::unwrap_failed
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library\core\src\option.rs:1985
   4: core::option::Option::unwrap
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6\library\core\src\option.rs:933
   5: gpui::platform::windows::platform::WindowsPlatform::close_one_window
             at .\src\platform\windows\platform.rs:139
   6: gpui::platform::windows::platform::impl$2::run
             at .\src\platform\windows\platform.rs:189
   7: gpui::app::App::run<hello_world::main::closure_env$0>
             at .\src\app.rs:140
   8: hello_world::main
             at .\examples\hello_world.rs:25
   9: core::ops::function::FnOnce::call_once<void (*)(),tuple$<> >
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6\library\core\src\ops\function.rs:250
  10: core::hint::black_box
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6\library\core\src\hint.rs:337
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
error: process didn't exit successfully: `V:\githubs\zed\target\debug\examples\hello_world.exe` (exit code: 101)
```

MS IME belike
![image](https://github.com/zed-industries/zed/assets/54507071/8cced28d-1868-471f-a042-ffc8e2eaf32a)

notice: [rime/weasel](https://github.com/rime/weasel) works fine on it

### Environment

```
Zed: v0.139.0 (Zed Dev 95e360b170d5cae4f18959b0c17e2fb2f9927a12)
OS: Windows 10.0.19045
Memory: 95.9 GiB
Architecture: x86_64
```

### If applicable, add mockups / screenshots to help explain present your vision of the feature

just panic on `thread 'main' panicked at crates\gpui\src\platform\windows\platform.rs:142:14:`

https://github.com/zed-industries/zed/blob/main/crates/gpui/src/platform/windows/platform.rs#L137-L143

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

panic without logs

**Comment by JunkuiZhang** [comment]
请问这个bug现在还有吗？

**Comment by shenjackyuanjie** [comment]
我看一眼去

**Comment by JunkuiZhang** [comment]
这bug也太离谱了，我的win10和win11一直都是正常的，也不知道有没有其他输入法能让我复现这个bug

**Comment by shenjackyuanjie** [comment]
确实能是修了
but, why?

**Comment by JunkuiZhang** [comment]
> 确实能是修了 but, why?

啊？不会崩溃了吗？

**Comment by shenjackyuanjie** [comment]
对的，这就修好了

**Comment by shenjackyuanjie** [comment]
testing fix by #15286

**Comment by shenjackyuanjie** [comment]
fixed by #15286
